cdap-hbase-compat-1.2-cdh5.7.0/src/main/java/co/cask/cdap/data2/util/hbase/HBase12CDH570TableUtil.java

```diff
@@ -102,17 +104,19 @@ public void deleteNamespaceIfExists(HBaseDDLExecutor ddlExecutor, String namespa
   }
 
   @Override
-  public void disableTable(HBaseAdmin admin, TableId tableId) throws IOException {
-    Preconditions.checkArgument(admin != null, "HBaseAdmin should not be null");
+  public void disableTable(HBaseDDLExecutor ddlExecutor, TableId tableId) throws IOException {
+    Preconditions.checkArgument(ddlExecutor != null, "HBaseDDLExecutor should not be null");
     Preconditions.checkArgument(tableId != null, "Table Id should not be null.");
-    admin.disableTable(nameConverter.toTableName(tablePrefix, tableId));
+    TableName tableName = nameConverter.toTableName(tablePrefix, tableId);
```

Comment:
just took a look at HTableNameConverter and I don't understand why its a concrete class and not a utility class with static methods. Can't disableTable and enableTable be the same for all versions of hbase? And if so, could we just remove those methods from HBaseTableUtil in favor of the ddl executor?